### ISSUE TITLE
Syntax error for "assert ... by ... "

### ISSUE THREAD

**Issue by orilahav** [ejgallego/coq-lsp#281]
assert ... by.. tactic generates "syntax error".
E.g. :

```coq
Lemma aaa : 1 = 0+1.
Proof.
assert (0+1 = 1) by (simpl; auto).
{ auto. }
Qed.
```

**Comment by Alizter** [comment]
Seems to work for me on both coq-lsp and CoqIDE. @orilahav Are you able to open other files? What is your lsp version / coq version?

**Comment by ejgallego** [comment]
@orilahav unfortunately this seems to be the problem that Coq parser gets indeed confused when opening multiple files; can you reproduce when you restart the server with just that file opened?

Next version will avoid this, maybe a good idea to release it ASAP.

**Comment by orilahav** [comment]
yes, this is unstable. I see this bug in long file, but the small example that failed before seems to work now.

coq                     8.16.0        Formal proof management system
coq-lsp                 0.1.4+v8.16   Language Server Protocol native server for

**Comment by ejgallego** [comment]
Thanks for the report @orilahav , this bug is fixed in Coq `master`, but I found a good workaround for the next release of `coq-lsp`, will try to update things ASAP.

For now if you get into this unfortunate error, the quickest way to solve it is keep just one .v file open, then run the "Restart coq-lsp server" command :( :(

**Comment by Alizter** [comment]
This is a bug in Coq that will be fixed in 8.17 which incorrectly backtracks when the parsing state is changed. So when one file loads something affecting the parsing of `assert .. by ..` the other files also see this since they are all related by backtracking.

**Comment by ejgallego** [comment]
@Alizter actually this is fixed for Coq 8.18, but I plan to try to backport the fix to 8.17, maybe we could even backport it to 8.16
